src/Nethermind/Nethermind.Core/Caching/LruCache.cs

```diff
@@ -27,7 +27,9 @@ public LruCache(int maxCapacity, int startCapacity, string name)
             _maxCapacity = maxCapacity;
             _cacheMap = typeof(TKey) == typeof(byte[])
                 ? new Dictionary<TKey, LinkedListNode<LruCacheItem>>((IEqualityComparer<TKey>)Bytes.EqualityComparer)
-                : new Dictionary<TKey, LinkedListNode<LruCacheItem>>(startCapacity); // do not initialize it at the full capacity
+                : (typeof(TKey) == typeof(ArraySegment<byte>)
+                    ? new Dictionary<TKey, LinkedListNode<LruCacheItem>>((IEqualityComparer<TKey>)Bytes.ArraySegmentEqualityComparer)
```

Comment:
We should move to SpanDictionary here too. Maybe through some Dictionary factory.

Reply:
Problem with LRU cache. The template for key cant be an array as it is also used for keccakkey. Need to modify it also to be able to use span dictionary with it.

Reply:
maybe just create SpanLruCache?